### PR TITLE
Enable JSR305 Null analysis in mvn and build groovy with jdt

### PR DESCRIPTION
**Pull request by triller-telekom**
Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

File: poms/tycho/pom.xml [hunk 1]
```diff
@@ -79,6 +79,9 @@
 				<version>${groovy.eclipse.compiler.version}</version>
 				<extensions>true</extensions>
 			</plugin>
+            <plugin>
```
Comment (kaikreuzer) [inline poms/tycho/pom.xml]: Could you replace spaces by tabs here?

File: poms/tycho/pom.xml [hunk 2]
```diff
@@ -79,6 +79,9 @@
 				<version>${groovy.eclipse.compiler.version}</version>
 				<extensions>true</extensions>
 			</plugin>
+			<plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
```
Comment (kaikreuzer) [inline poms/tycho/pom.xml]: What about the spaces in this line?

File: poms/tycho/pom.xml [hunk 3]
```diff
@@ -79,6 +79,9 @@
...
+            </plugin>
```
Comment (kaikreuzer) [inline poms/tycho/pom.xml]: ... and in that?

**Comment by martinvw** [comment]
Is is related to the breaking of the builds in openHAB 

> [ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:1.0.0:compile (default-compile) on project org.openhab.binding.amazondashbutton: Compilation failure: Compilation failure:
[ERROR] /home/travis/build/openhab/openhab2-addons/addons/binding/org.openhab.binding.amazondashbutton/src/main/java/org/openhab/binding/amazondashbutton/internal/pcap/PcapNetworkInterfaceWrapper.java:[30]
[ERROR] public class PcapNetworkInterfaceWrapper {
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The method equals(Object) from Object cannot implement the corresponding method from Function<PcapNetworkInterface,PcapNetworkInterfaceWrapper> due to incompatible nullness constraints
[ERROR] /home/travis/build/openhab/openhab2-addons/addons/binding/org.openhab.binding.amazondashbutton/src/main/java/org/openhab/binding/amazondashbutton/internal/pcap/PcapNetworkInterfaceWrapper.java:[43]
[ERROR] public PcapNetworkInterfaceWrapper apply(PcapNetworkInterface pcapNetworkInterface) {
[ERROR] ^^^^^^^^^^^^^^^^^^^^
[ERROR] Missing nullable annotation: inherited method from Function<PcapNetworkInterface,PcapNetworkInterfaceWrapper> specifies this parameter as @Nullable
[ERROR] /home/travis/build/openhab/openhab2-addons/addons/binding/org.openhab.binding.amazondashbutton/src/main/java/org/openhab/binding/amazondashbutton/internal/pcap/PcapNetworkInterfaceService.java:[1]
[ERROR] /**
[ERROR] ^
[ERROR] The method equals(Object) from Object cannot implement the corresponding method from Predicate<PcapNetworkInterfaceWrapper> due to incompatible nullness constraints
[ERROR] /home/travis/build/openhab/openhab2-addons/addons/binding/org.openhab.binding.amazondashbutton/src/main/java/org/openhab/binding/amazondashbutton/internal/pcap/PcapNetworkInterfaceService.java:[177]
[ERROR] public boolean apply(PcapNetworkInterfaceWrapper networkInterface) {
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] Missing nullable annotation: inherited method from Predicate<PcapNetworkInterfaceWrapper> specifies this parameter as @Nullable
[ERROR] 4 problems (4 errors)

**Comment by martinvw** [comment]
See for example https://travis-ci.org/openhab/openhab2-addons/builds/260086842#L468

A build result of https://github.com/openhab/openhab2-addons/pull/2380#event-1189024056

**Comment by martinvw** [comment]
And the master build fails also now:

https://openhab.ci.cloudbees.com/job/openHAB2-Bundles/107/

**Comment by martinvw** [comment]
@triller-telekom @kaikreuzer ^^

**Comment by triller-telekom** [comment]
Sorry, i forgot one parameter for maven, see https://github.com/openhab/openhab-core/pull/174

**Pull request merged**